internal/repository/repository.go

```diff
@@ -43,28 +50,167 @@ type ConversionRequest struct {
 }
 
 // Repository represents the layer between the business logic and the database.
-type Repository struct{}
+type Repository struct {
+	db *sql.DB
+}
 
-func NewRepository() *Repository {
-	return &Repository{}
+func NewRepository(db *sql.DB) *Repository {
+	return &Repository{db: db}
 }
 
 // InsertUser inserts the user into users table and returns user id.
 func (r *Repository) InsertUser(email, password string) (string, error) {
-	return "", nil
+	var userID string
+	const query = "insert into converter.users (email, password) values ($1, $2) returning id;"
+
+	err := r.db.QueryRow(query, email, password).Scan(&userID)
+	if err, ok := err.(*pq.Error); ok {
+		if err.Code == uniqueViolationCode {
+			return "", ErrUserAlreadyExists
+		} else {
+			return "", fmt.Errorf("can't insert user: %s", err.Code.Name())
+		}
+	}
```

Comment:
Let's flatten this error handling to check the first case is uniqueViolationCode and all other errors as a second case